dev/release/verify-release-candidate.sh

```diff
@@ -693,6 +702,8 @@ test_linux_wheels() {
     local arch="x86_64"
   fi
 
+  local declared_package_version="${VERSION_NATIVE}"
+  local package_version="${TEST_PYARROW_VERSION:-${declared_package_version}}"
```

Comment:
```suggestion
  local package_version="${TEST_ADBC_VERSION:-${declared_package_version}}"
```

Or `TEST_ADBC_PYTHON_VERSION`.

(I think that we can always use `${VERSION_NATIVE}` here.)

Reply:
Changed to always use VERSION_NATIVE